assets/reader-activation/auth.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Internal dependencies.
+ */
+import './auth.scss';
+
+( function ( readerActivation ) {
+	window.onload = function () {
+		if ( ! readerActivation ) {
+			return;
+		}
+
+		const container = document.querySelector( '#newspack-reader-auth' );
+		if ( ! container ) {
+			return;
+		}
+
+		const initialForm = container.querySelector( 'form' );
+		let form;
+		/** Workaround AMP's enforced XHR strategy. */
+		if ( initialForm.getAttribute( 'action-xhr' ) ) {
+			initialForm.removeAttribute( 'action-xhr' );
+			form = initialForm.cloneNode( true );
+			initialForm.replaceWith( form );
+		} else {
+			form = initialForm;
+		}
+
+		const actionInput = form.querySelector( 'input[name="action"]' );
+		const emailInput = form.querySelector( 'input[name="email"]' );
+		const passwordInput = form.querySelector( 'input[name="password"]' );
+		const submitButtons = form.querySelectorAll( '[type="submit"]' );
+
+		container.querySelector( 'button[data-close]' ).addEventListener( 'click', function ( ev ) {
+			ev.preventDefault();
+			container.style.display = 'none';
+		} );
+
+		const messageContainer = container.querySelector( '.newspack-reader__auth-form__response' );
+
+		const authLinkMessage = container.querySelector( '[data-has-auth-link]' );
+		authLinkMessage.hidden = true;
+
+		const accountLinks = document.querySelectorAll( '.newspack-reader__account-link' );
+		const triggerLinks = document.querySelectorAll( '[data-newspack-reader-account-link]' );
+
+		/**
+		 * Handle reader changes.
+		 */
+		readerActivation.on( 'reader', ( { detail: { email, authenticated } } ) => {
```

Comment:
`authenticated` is always `undefined` in my testing, regardless of user being new or existing.

Reply:
`undefined` is falsy so should be interpreted as not authenticated. 32cffd9314257290bab489de064f47a10bda5ccc makes sure to set it to false per backend response.

Reply:
I was expecting registration via the Registration block to handle the `authenticated` part here. But we're missing a `readerActivation.setAuthenticated( data.authenticated )` in the [block's `endLoginFlow` handler](https://github.com/Automattic/newspack-plugin/blob/6bb2d1c66ae724d58a9ae6bd210539d6c763bcf6/assets/blocks/reader-registration/view.js#L30).

Reply:
Good call! I don't know why I didn't address it on this PR, it's the whole purpose of the frontend library 😅 

With ae449b5e2e7a561e4fccded79564cd207f4d8834 you can now see the reader registration block changing the account link label and click behavior 🎉